GoogleForCreators/web-stories-wp
# UI Simplification: Allow selecting Layer from the element context menu

## Context


## Summary
Adds a new item to the Context Menu (right-click menu of the element): Select Layer


## Relevant Technical Choices

- a11y: Since moving between parent-submenu happens with arrow keys, using left and right arrow keys does not work anymore for moving vertically to the next item in the menu,
- Using the same `Menu` component for rendering the submenu as well.



## To-do

- [x] Don't display Select Layer if empty results for intersecting elements
- [x] Accessibility -- Left-Right arrow keys will be only for moving between submenu and the parent menu now.
- [x] When opening the context menu from keyboard, what to display then?
- ~Refactor the right-click menu file into smaller files.~ Will be done by Pea Pod in https://github.com/google/web-stories-wp/issues/10212 instead.
- [x] Style details to make it match with Figma more precisely (adjust paddings + add Select Layer icon)



## User-facing changes
<img width="499" alt="Screenshot 2022-01-17 at 16 01 55" src="https://user-images.githubusercontent.com/3294597/149782482-4fe60bf7-2a56-4e92-af44-cb0d87f35b06.png">



## Testing Instructions





This PR can be tested by following these steps:

1. Add some elements.
2. Now add an image on top of the other elements.
3. Open the right-click menu for the image.
4. Verify it has "Select Layer" option, click on it.
5. Verify you can see the top layer (image) selected and other layers listed **that were under the place where you clicked with cursor when opening the right-click menu.** 
6. Verify that selecting a layer in the submenu will select it and close the menu.
7. Test the same for all other element types as well (single-selection only!)
8. Verify that opening the right-click menu from the Layer panel will not display the Select Layer item in the menu (since there are no elements under the cursor while doing it!)
9. Verify that opening the right-click menu on top of one element only will not display the Select Layer item in the menu (since the only element is already selected, so no point opening it!)
10. Verify that layers that are not under the cursor while opening the right-click menu will not be displayed in the submenu either.
11. Accessibility: verify you can move between the parent menu and the submenu using arrow keys.


## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
Yes, tracking `context_menu_action` + `layer_selected` now.


### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10107
